dev/src/serializer.ts

```diff
@@ -286,6 +289,75 @@ export class Serializer {
       }
     }
   }
+
+  /**
+   * Decodes a google.protobuf.Value
+   *
+   * @private
+   * @internal
+   * @param proto A Google Protobuf 'Value'.
+   * @returns The converted JS type.
+   */
+  decodeGoogleProtobufValue(proto: IValue): unknown {
+    switch (detectGoogleProtobufValueType(proto)) {
+      case 'nullValue': {
+        return null;
+      }
+      case 'numberValue': {
+        return proto.numberValue;
+      }
+      case 'stringValue': {
+        return proto.stringValue;
+      }
+      case 'boolValue': {
+        return proto.boolValue;
+      }
+      case 'listValue': {
+        const array: unknown[] = [];
+        if (Array.isArray(proto.listValue!.values)) {
+          for (const value of proto.listValue!.values!) {
+            array.push(this.decodeGoogleProtobufValue(value));
+          }
+        }
+        return array;
+      }
+      case 'structValue': {
+        const obj: DocumentData = {};
+        const fields = proto.structValue?.fields;
+        if (fields) {
+          for (const prop of Object.keys(fields)) {
+            obj[prop] = this.decodeGoogleProtobufValue(fields[prop]);
+          }
+        }
```

Comment:
nit: there's an opportunity to re-use code in `decodeGoogleProtobufStruct`.

Reply:
Done